src/pkg/services/m365/api/events.go

```diff
@@ -153,24 +153,30 @@ func (c Events) GetContainerByName(
 		return nil, graph.Stack(ctx, err).WithClues(ctx)
 	}
 
-	// We only allow the api to match one calendar with provided name.
+	gv := resp.GetValue()
+
+	if len(gv) == 0 {
+		return nil, clues.New("container not found").WithClues(ctx)
+	}
+
+	// We only allow the api to match one calendar with the provided name.
 	// Return an error if multiple calendars exist (unlikely) or if no calendar
 	// is found.
 	if len(resp.GetValue()) != 1 {
```

Comment:
is this going to block restores for users that already have multiple calendars with the same name?

Reply:
ooh, good call.  I suppose we're best off just taking the first of many.